### ISSUE TITLE
[Regression] PartDesign: 'Mirroring' breaks both the original and the mirror of a specific PD (RC2)

### ISSUE THREAD

**Issue by gdecarpentier** [FreeCAD/FreeCAD#17150]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem description

I tried to mirror a reasonably complex part design (that was largely built using v0.21 and then I upgraded to RC2) and that created edges in wrong new directions in RC2. I've attached a ZIP file containing the project with the part and the video showing the repro. (An analogous mirror operation in v0.21 didn't seem to have this problem on this part). Hope this helps!

### Full version info

```shell
OS: Windows 10 build 19045
Word size of FreeCAD: 64-bit
Version: 1.0.0RC2.38806 (Git)
Build type: Release
Branch: (HEAD detached at 1.0rc2)
Hash: 3d63fc6c2f665a8d5e6468845a419bcac80756c7
Python 3.11.9, Qt 5.15.13, Coin 4.0.3, Vtk 9.2.6, OCC 7.7.2
Locale: English/United States (en_US)
Stylesheet/Theme/QtStyle: FreeCAD Dark.qss/FreeCAD Dark/Fusion
```

### Subproject(s) affected?

PartDesign

### Anything else?

Screecast:  
https://github.com/user-attachments/assets/501de3c6-7c80-4007-af66-bf2f69cb56b7
Example file: [mirroring_bug.zip](https://github.com/user-attachments/files/17387605/mirroring_bug.zip)

Includes the screencast (if it doesn't display): [mirroring_bug.zip](https://github.com/user-attachments/files/17317717/mirroring_bug.zip)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by maxwxyz** [comment]
Please provide an example file.
@NomAnor FYI

**Comment by gdecarpentier** [comment]
File and repro are already in the zip

**Comment by gdecarpentier** [comment]
@maxwxyz Can you tell me why you say it needs an example file and repro steps when both are included?

**Comment by AIRCAP** [comment]
can be fixed by #17119

**Comment by AIRCAP** [comment]
The demo part only breaks if the mirror operation is refined. if its not refined it works and passes BOP checks. I assume at the time of reporting the bug 0.22 had Refinement turned on by default. it has since been turned off again unless set otherwise in preferences
